src/scanner.ts

```diff
@@ -392,14 +392,19 @@ export class Scanner {
             }
         }
 
-        return this.source.slice(start, this.index);
+        return {
+            escapedChars: 0,
```

Comment:
Oh, true. In this case it'd be possible to anotate the number of escaped characters in some property at the scanner class, or a variable.